File: app/src/main/java/me/echeung/moemoekyun/api/services/UsersService.java
package me.echeung.moemoekyun.api.services;

import me.echeung.moemoekyun.api.ErrorHandlingAdapter;
import me.echeung.moemoekyun.api.responses.UserResponse;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;

public interface UsersService {
    @GET("users/{username}")
    ErrorHandlingAdapter.WrappedCall<UserResponse> getUserInfo(@Header("Authorization") String token, @Path("username") String username);
}

File: app/src/main/java/me/echeung/moemoekyun/client/api/v4/service/UsersService.kt
package me.echeung.moemoekyun.client.api.v4.service

import me.echeung.moemoekyun.client.api.v4.ErrorHandlingAdapter
import me.echeung.moemoekyun.client.api.v4.response.UserResponse
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Path

interface UsersService {
    @GET("users/{username}")
    fun getUserInfo(@Header("Authorization") token: String, @Path("username") username: String): ErrorHandlingAdapter.WrappedCall<UserResponse>
}
